backend/vnc.c

```diff
@@ -0,0 +1,246 @@
+    size_t span_width = right - left;
+
+    memcpy(fb_pixels, pixels, span_width * sizeof(uint32_t));
+
+    pixman_region_init_rect(&tx->damage_region, left, top, span_width, 1);
+
+    if (pixman_region_not_empty(&tx->damage_region)) {
+        nvnc_display_feed_buffer(tx->display, tx->current_fb,
+                                 &tx->damage_region);
+        pixman_region_clear(&tx->damage_region);
+    }
+    aml_poll(tx->aml, 0);
+    aml_dispatch(tx->aml);
+}
+
+static void twin_vnc_get_screen_size(twin_vnc_t *tx, int *width, int *height)
+{
+    *width = nvnc_fb_get_width(tx->current_fb);
+    *height = nvnc_fb_get_height(tx->current_fb);
+}
+
+static bool _twin_vnc_work(void *closure)
+{
+    twin_screen_t *screen = SCREEN(closure);
+
+    if (twin_screen_damaged(screen))
+        twin_screen_update(screen);
+    return true;
+}
+
+static void _twin_vnc_new_client(struct nvnc_client *client)
+{
+    twin_peer_t *peer = malloc(sizeof(twin_peer_t));
+    nvnc_set_userdata(client, peer, NULL);
+}
+
+static bool _twin_vnc_read_events(int fd, twin_file_op_t op, void *closure)
+{
+    return true;
+}
+
+static void _twin_vnc_pointer_event(struct nvnc_client *client,
+                                    uint16_t x,
+                                    uint16_t y,
+                                    enum nvnc_button_mask button)
+{
+    twin_peer_t *peer = nvnc_get_userdata(client);
+    twin_event_t tev;
+    if ((button & NVNC_BUTTON_LEFT) &&
+        !(peer->prev_button & NVNC_BUTTON_LEFT)) {
+        tev.u.pointer.screen_x = x;
+        tev.u.pointer.screen_y = y;
+        tev.kind = TwinEventButtonDown;
+        tev.u.pointer.button = 1;
+    } else if (!(button & NVNC_BUTTON_LEFT) &&
+               (peer->prev_button & NVNC_BUTTON_LEFT)) {
+        tev.u.pointer.screen_x = x;
+        tev.u.pointer.screen_y = y;
+        tev.kind = TwinEventButtonUp;
+        tev.u.pointer.button = 1;
+    }
+    if ((peer->px != x || peer->py != y)) {
+        peer->px = x;
+        peer->py = y;
+        tev.u.pointer.screen_x = x;
+        tev.u.pointer.screen_y = y;
+        tev.u.pointer.button = 0;
+        tev.kind = TwinEventMotion;
+    }
+    peer->prev_button = button;
+    struct nvnc *server = nvnc_client_get_server(client);
+    twin_vnc_t *tx = nvnc_get_userdata(server);
+    twin_screen_dispatch(tx->screen, &tev);
+}
+
+
+twin_context_t *twin_vnc_init(int width, int height)
+{
+    twin_context_t *ctx = calloc(1, sizeof(twin_context_t));
+    if (!ctx)
+        return NULL;
+
+    ctx->priv = calloc(1, sizeof(twin_vnc_t));
+    if (!ctx->priv) {
+        free(ctx);
+        return NULL;
+    }
+
+    twin_vnc_t *tx = ctx->priv;
+    tx->width = width;
+    tx->height = height;
+
+    tx->aml = aml_new();
+    if (!tx->aml)
+        goto bail_priv;
+    aml_set_default(tx->aml);
+    tx->server = nvnc_open(VNC_HOST, VNC_PORT);
+    if (!tx->server)
+        goto bail_aml;
```

Comment:
Print error messages for all similar cases.
Check: https://github.com/sysprog21/mado/blob/main/backend/fbdev.c#L104